lib/elixir/lib/exception.ex

```diff
@@ -1342,8 +1342,21 @@ defmodule UndefinedFunctionError do
       hint_for_loaded_module(module, function, arity, nil)
   end
 
-  defp hint(_module, _function, _arity, _loaded?) do
-    ""
+  defp hint(module, function, arity, _loaded?) do
+    {module, _distance} =
+      Enum.map(:code.all_available(), fn {name, _, _} ->
+        {name, String.jaro_distance(to_string(module), to_string(name))}
+      end)
+      |> Enum.sort_by(&elem(&1, 1), :desc)
+      |> Enum.take(3)
```

Comment:
I think you probably want a threshold first like in [`Exception`](https://github.com/elixir-lang/elixir/blob/main/lib/elixir/lib/exception.ex#L1385), and also handle the empty case.

Sorry just thinking out loud, I also wonder if we could do something like:
- score modules with jaro distance
- filter on threshold
- score module.function strings based on jaro distance
- re-filter on threshold, sort & take

(in pseudo-code)
```elixir
for candidate_mod <- :code.available(),
  mod_score = jaro(module, candidate_mod),
  mod_score >= @threshold,
  candidate_fun <- funs_in_module(candidate_mod),
  mod_fun_score = jaro(module, fun, candidate_mod, candidate_fun),
  mod_fun_score >= @threshold,
  do: {candidate_mod, candidate_fun, mod_fun_score}
|> sort_by...
|> take...
```

Reply:
Yep it will end up being a version of this! The initial implementation was just a POC